### PR TITLE
[ADD] hr_cae_documents: new module

### PR DESCRIPTION
**Pull request by robinkeunen**
* copy documents from applicant to employee
* add expiration date to attachments
* add dependency in coopaname_custom

### CODE REVIEW

**Comment by robinkeunen** [comment]
ben ... il passe mon build nondidjap, jvais faire un rebase pour retrigger

**Review by quentinDupont — APPROVED** [review]
Works fine :white_check_mark: 
Will not be used in GRAP so i didn't test with my colleague

File: hr_cae_documents/models/ir_attachment.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2019 Coop IT Easy SCRL fs
+#   Robin Keunen <robin@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class AttachmentCategory(models.Model):
+    _name = "ir.attachment.category"
+    # move to separate module?
```
Comment (mclaeysb) [inline hr_cae_documents/models/ir_attachment.py:11]: Can't think of a OCA repo to put this in. Would keep this here for now.

File: hr_cae_documents/security/ir.model.access.csv [hunk 2]
```diff
@@ -0,0 +1,3 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_ir_attachment_category_user,access_ir_attachment_category_user,model_ir_attachment_category,base.group_user,1,0,0,0
```
Comment (mclaeysb) [inline hr_cae_documents/security/ir.model.access.csv:2]: I'm quite surprised to see that users can't delete attachements 😉

File: hr_cae_documents/models/hr_applicant.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    def create_employee_from_applicant(self):
+        employee_action_window = super().create_employee_from_applicant()
+        # note: this function does not use `ensure_one`. It returns the
+        # action window on the last created employee.
+
+        for applicant in self:
+            employee = applicant.emp_id
+            for document in applicant.attachment_ids:
+                employee_document = document.copy()
+                employee_document.res_model = employee._name
```
Comment (mclaeysb) [inline hr_cae_documents/models/hr_applicant.py:21]: How does this ensure that the document will be visible on the employee page? Employee's don't seem to have attachements or an action button.

**Review by mclaeysb — COMMENTED** [review]
Interesting code, had some small questions

File: hr_cae_documents/security/ir.model.access.csv (hunk 2 above)
Comment (robinkeunen) [reply hr_cae_documents/security/ir.model.access.csv:2]: They can't delete attachement **category**!

**Pull request merged**